net-im/synapse: add 1.65.0

arkamar opened the pull request:
Straight forward version bump, all tests passed, there is just small difference between this ebuild and `1.63.1`, see the comment https://github.com/gentoo/gentoo/pull/26706#issuecomment-1216983750 below.

BratishkaErik commented:
https://matrix.org/blog/2022/08/03/synapse-1-64-released#memory-leak-in-frozendict

> A couple of weeks ago, we identified a memory leak within frozendict, which is a library that Synapse relies on. This would in turn cause Synapse instances to slowly leak memory when processing /sync requests.
> We highly encourage server administrators who installed Synapse via pip to upgrade their local version of frozendict to version 2.3.3 or later, which includes a fix to this issue. 

So it depends on https://github.com/gentoo/gentoo/pull/26692

arkamar commented:
New version `1.65.0` was released few hours ago. I have updated this PR to this version and rebased on top of current `master` branch. There is a small change in the ebuild from previous version, it newly inherites `multiprocessing` eclass in order to run tests in parallel with `twisted.trial -j "$(makeopts_jobs)"` parameter.
